x-pack/plugins/alerting/server/lib/process_alerts.ts

```diff
@@ -123,7 +132,7 @@ function processAlertsHelper<
             updateAlertFlappingHistory(activeAlerts[id], false);
           }
         }
-      } else if (existingAlertIds.has(id)) {
+      } else if (existingAlertIds.has(id) && autoRecoverAlerts) {
```

Comment:
Skips this step for persistent alert rule types (slight time optimization when running a rule)